code-challenges.test.js

```diff
@@ -56,6 +96,41 @@ var hand2 = [5, 5, 3, 3, 4]
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
 
-
-
+//TEST:
+    //describe "fullHouse"
+    //it "takes in an array of 5 numbers and determines whether or not the array is a 'full house'"
+    //expect hand1 to equal true
+    //expect hand2 to equal false
+    //expect hand3 to equal false
+    
+    describe("fullHouse", () => {
+        it("takes in an array of 5 numbers and determines whether or not the array is a 'full house'", () => {
+            expect(fullHouse(hand1)).toEqual(true)
+            expect(fullHouse(hand2)).toEqual(false)
+            expect(fullHouse(hand3)).toEqual(false)
+        })
+    })
 // b) Create the function that makes the test pass.
+//PROCESS:
+    //declare a function- fullHouse
+    //parameter- array
+    //iteration- for loop to check for pairs and three of one kind in each index of the array
+    //catchall- to make sure input array is 5 numbers
+    //inputs and outputs- hand1 --> true, hand2 --> false, hand3 --> false
+    
+    const fullHouse = (array) => {
+        for(let i=0; i<5; i++){
+            if(array[0] === array[1] && array[0] === array[2] && array[3] === array[4]){
+                return true
+            }else if(array[0] === array[4] && array[1] === array[2] && array[1] === array[3]){
+                return true
+            }else if(array[0] === array[1] && array[2] === array[3] && array[2] === array[4]){
+                return true
+            }else if(array[0] === array[2] && array[2] === array[4] && array[1] === array[3]){
+                return true  
+            }else if(array.length != 5){
+                return "Please enter 5 numbers"
+            }else return false
+        }
+    }
+//This was a hard one for me! I think it works for the given variables but I know there are more combinations of numbers that could possibly make my test fail. I couldn't figure out an easier/more dynamic way to make this work🤔 hmm..
```

Comment:
This is a great place to be. It does the job but you know there are ways it could be refactored. As you learn about different ideas and tools you could come back to this problem and try it again!